documentation/modules/ROOT/pages/transformations/mongodb-event-flattening.adoc

```diff
@@ -9,57 +13,163 @@
-This SMT is supported only for the MongoDB connector.
-See xref:{link-event-flattening}[Extracting source record `after` state from {prodname} change events] for the relational database equivalent to this SMT.
+The information in this chapter describes the event flattening single message transformation (SMT) for {prodname} MongoDB connectors only.
+For information about an equivalent SMT for use with relational databases, see the {link-prefix}:{link-event-flattening}#new-record-state-extraction[documentation for the New Record State Extraction SMT].
 ====
 
-The {prodname} MongoDB connector generates the data in a form of a complex message structure.
-The message consists of two parts:
+ifdef::product[]
+The following topics provide details:
+
+* xref:description-of-debezium-mongodb-change-event-structure[]
+* xref:behavior-of-debezium-mongodb-event-flattening-transformation[]
+* xref:configuration-of-the-debezium-mongodb-event-flattening-transformation[]
+* xref:options-for-encoding-arrays-in-mongodb-event-messages[]
+* xref:flattening-nested-structures-in-a-mongodb-event-message[]
+* xref:how-the-debezium-mongodb-connector-reports-the-names-of-fields-removed-by-unset-operations[]
+* xref:mongodb-event-flattening-determining-the-type-of-the-original-database-operation[]
+* xref:using-the-mongodb-event-flattening-smt-to-add-debezium-metadata-to-kafka-records[]
+* xref:options-for-applying-the-mongodb-extract-new-document-state-transformation-selectively[]
+* xref:configuration-options-for-the-debezium-mongodb-event-flattening-transformation[]
+* xref:debezium-event-flattening-smt-for-mongodb-known-limitations[]
+endif::product[]
 
-* operation and metadata
-* for inserts, the whole data after the insert has been executed; for updates a patch element describing the altered fields
+// Type: concept
+// ModuleID: description-of-debezium-mongodb-change-event-structure
+// Title: Description of {prodname} MongoDB change event structure
+== Change event structure
+
+The {prodname} MongoDB connector generates change events that have a complex structure.
+Each event message includes the following parts:
+
+Source metadata:: Includes, but is not limited to the following fields:
+
+* Type of the operation that changed data in the collection (insert, update, or delete).
+* Name of the database and collection in which the change occurred.
+* Timestamp that identifies when the change was made.
+* Optional transaction information.
+
+Document data::
+`before` data:::
+This field is present only when `capture.mode` is set to one of the following values:
+** `change_streams_with_pre_image`
+** `change_streams_update_full_with_pre_image`.
+
+`after` data:::
+JSON strings that represent the values that are present in a document after the current operation.
+The presence of an `after` field in an event message depends on the type of event and the connector configuration.
+`insert` events always contain an `after` field regardless of the `capture.mode` setting.
```

Comment:
One larger question - we use both `create` and `insert` in this document and also in MongoDB documentation.
I wonder if it is a bit misleading. In the text itself it is probably not an issue but the technical name of event should be unified to `create`

Reply:
Thanks for pointing that out. I think that the MongoDB terminology talks about `insert` operations, so I changed some instances to reflect that, but I'll review and make sure that any place that talks about Debezium events uses the `create` terminology.